src/data/polynomial/reverse.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.degree.basic
+import data.polynomial.erase_lead
+import data.polynomial.degree.trailing_degree
+
+/-!
+# Reverse of a univariate polynomial
+
+The main definition is `reverse`.  Applying `reverse` to a polynomial `f(x)` produces the
+polynomial `x^f.nat_degree * f(1/x)`.
+
+The main result is that `reverse (f * g) = reverse (f) * reverse (g)`, provided the leading
+coefficients of `f` and `g` do not multiply to zero.
+-/
+
+namespace polynomial
+
+open polynomial finsupp finset --erase_lead
+
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+
+namespace rev
+/-- rev_at is a function of two natural variables (N,i).  If i ≤ N, then rev_at N i returns N-i,
+otherwise it returns N.  Essentially, this function is only used for i ≤ N. -/
+def rev_at (N : ℕ) : ℕ → ℕ := λ i : ℕ, ite (i ≤ N) (N-i) i
+
+@[simp] lemma rev_at_invol {N n : ℕ} : rev_at N (rev_at N n) = n :=
+begin
+  unfold rev_at,
+  split_ifs with h j,
+  { exact nat.sub_sub_self h, },
+  { exfalso,
+    apply j,
+    exact nat.sub_le N n, },
+  { refl, },
+end
+
+@[simp] lemma rev_at_small {N n : ℕ} (H : n ≤ N) : rev_at N n = N-n :=
```

Comment:
I see a few `@[simp]` lemmas that require proofs not occuring in the equalities: `rev_at_small`,  `reflect_mul`. I think those will not be picked up by the `simp` tactic because it cannot figure out what to fill in for the proofs. Have you tested that they work?

Reply:
I am afraid that I do not understand this question, and hence the answer is no, I guess.  I do not know how ```simp``` works, but my intention was to communicate to Lean that ```rev_at``` should make it think of splitting the two cases that appear in the definition.  What tests should I do?